app/controllers/ticket_purchases_controller.rb

```diff
@@ -4,32 +4,22 @@ class TicketPurchasesController < ApplicationController
   authorize_resource :conference_registrations, class: Registration
 
   def create
+    TicketPurchase.by_conference(@conference).unpaid.by_user(current_user).destroy_all
```

Comment:
Why not 

``
current_user.ticket_purchases.by_conference(@conference).unpaid.destroy_all
``